build-libs/redirects.js

```diff
@@ -135,63 +135,72 @@ async function getLatestContentRefForProduct(product) {
+ *
+ * In all other cases, we're building from a content repository that does not
+ * relate to the redirects we're trying to fetch. So, we can safely ignore the
+ * redirects and return early with an empty array.
+ *
+ * @param {string} repoName The name of the repo, owner is always `hashicorp`.
+ * @param {string?} redirectsPath Optionally specify a custom path to the redirects file.
+ * @returns {Promise<Redirect[]>}
  */
-async function getRedirectsForProduct(
-	/** @type {string} The product slug. Corresponds to a repository name. */
-	product,
-	{ ref = 'stable-website', redirectsPath = 'website/redirects.js' } = {}
+async function getRedirectsFromContentRepo(
+	repoName,
+	redirectsPath = 'website/redirects.js'
 ) {
-	let latestRef = ref
-	try {
-		latestRef = await getLatestContentRefForProduct(product)
-	} catch (error) {
-		// do nothing
-		console.warn(
-			'[redirects] failed to fetch latestRef for:',
-			product,
-			error.message
-		)
-	}
-
-	let repo = product
-
-	/** @type {string} A raw redirects file string to evaluate */
-	let rawRedirects
-
 	/**
-	 * Load the raw redirects
+	 * Note: These constants are declared for clarity in build context intent.
 	 */
-	if (isDeployPreview(product)) {
-		// For deploy previews of this product, load redirects locally,
-		// as authors may be modifying redirects as part of their work.
-		rawRedirects = fs.readFileSync(
-			path.join(process.cwd(), '../redirects.js'),
-			'utf-8'
-		)
-	} else if (isDeployPreview()) {
-		// For deploy previews in "other products", return an empty array,
-		// since we'll be removing the content for this product anyways.
-		rawRedirects = '[]'
-	} else {
-		// Otherwise, such as for production deploys,
-		// load redirects from the product repository.
-		rawRedirects = await fetchGithubFile({
+	const isDeveloperBuild = !process.env.IS_CONTENT_PREVIEW
+	const isLocalContentBuild = isDeployPreview(repoName)
+	/**
+	 * Load redirects from the target repo (or return early for non-target repos).
+	 */
+	/** @type {string} */
+	let redirectsFileString
+	if (isDeveloperBuild) {
+		// For `hashicorp/dev-portal` builds, load redirects remotely
+		const latestContentRef = await getLatestContentRefForProduct(repoName)
+		redirectsFileString = await fetchGithubFile({
 			owner: 'hashicorp',
-			repo,
+			repo: repoName,
 			path: redirectsPath,
-			ref: latestRef ?? ref,
+			ref: latestContentRef,
 		})
+	} else if (isLocalContentBuild) {
+		// Load redirects from the filesystem, so that authors can see their changes
+		const redirectsFilePath = path.join(process.cwd(), '../redirects.js')
+		redirectsFileString = fs.readFileSync(redirectsFilePath, 'utf-8')
+	} else {
+		// Return early, in this build we can ignore repoName's redirects.
+		return []
 	}
-
+	/**
+	 * Evaluate the redirects file string, filter invalid redirects, and add
+	 * a host condition for proxied sites.
+	 *
+	 * TODO: remove `addHostCondition` once Sentinel is migrated (ie once
```

Comment:
Request: Add your name to TODOs — the idea is not that you will necessarily be the one to address this TODO in the future, it's moreso an indicator of "who authored this TODO".

This is a nicer alternative to `git blame` / `git history`, which sometimes gets obfuscated by multi-line changes, and directs folks to incorrect TODO-authors.

```suggestion
	 * TODO(zachshilton): remove `addHostCondition` once Sentinel is migrated (ie once
```

Reply:
The red diff here is an example of me tagging myself in TODOs
https://github.com/hashicorp/dev-portal/pull/2128/commits/5499c850cef0ebd1de2320c5594b427b5fe9ad28

Reply:
ooh nice i love this pattern!